File: leetcode/1456.maximum-number-of-vowels-in-a-substring-of-given-length.cpp
/*
 * @lc app=leetcode id=1456 lang=cpp
 *
 * [1456] Maximum Number of Vowels in a Substring of Given Length
 */

// @lc code=start
#include <algorithm>
class Solution {
public:
    int maxVowels(string s, int k) {
        char vowels[5] = {'a', 'u', 'o', 'i', 'e'};
        int vowelCount = 0;
        for (int i=0; i<k; i++) {
            if (std::find(std::begin(vowels), std::end(vowels), s[i]) != std::end(vowels)) {
                vowelCount++;
            }
        }

        int maxCount = vowelCount;


        for (int i=k; i<s.size(); i++) {
            if (std::find(std::begin(vowels), std::end(vowels), s[i-k]) != std::end(vowels)) {
                vowelCount--;
            }
            if (std::find(std::begin(vowels), std::end(vowels), s[i]) != std::end(vowels)) {
                vowelCount++;
            }

            maxCount=std::max(maxCount, vowelCount);

        }

        return maxCount;



    }
};
// @lc code=end

File: leetcode/1456.maximum-number-of-vowels-in-a-substring-of-given-length.py
#
# @lc app=leetcode id=1456 lang=python3
#
# [1456] Maximum Number of Vowels in a Substring of Given Length
#

# @lc code=start
class Solution:
    def maxVowels(self, s: str, k: int) -> int:
        vowels_list = ["a", "e", "i", "o", "u"]
        count=0
        for i in range(k):
            if s[i] in vowels_list:
                count+=1
        
        maxCount = count

        for i in range(k, len(s)):
            if s[i] in vowels_list:
                count+=1
            
            if s[i-k] in vowels_list:
                count-=1
            
            maxCount = max(count, maxCount)
        

        return maxCount
# @lc code=end

